### PR TITLE
Add mutation testing

### PR DESCRIPTION
**Pull request by WillGibson**
As discussed here: https://github.com/mariosimao/notion-sdk-php/pull/6#discussion_r745935923

Some comments inline in the diff.

It includes fixing some mutations which were all the same thing so I could check out how it worked etc. I'd be happy to split those off to a separate pull request if preferred.

### CODE REVIEW

File: infection.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        ]
+    },
+    "logs": {
+        "text": ".infection/infection.log",
+        "summary": ".infection/summary.log",
+        "json": ".infection/infection-log.json",
+        "perMutator": ".infection/per-mutator.md"
+    },
+    "minMsi": 81,
+    "minCoveredMsi": 89,
```
Comment (WillGibson) [inline infection.json:15]: These were arbitrary numbers I chose so I could see it fail the CI step then make it pass without having to make loads of changes.

File: src/Blocks/Bookmark.php [hunk 2]
```diff
@@ -34,7 +35,7 @@ class Bookmark implements BlockInterface
     private function __construct(Block $block, string $url, array $caption)
     {
         if (!$block->isBookmark()) {
-            throw new \Exception("Block must be of type " . self::TYPE);
+            throw new BlockTypeException(self::TYPE);
```
Comment (WillGibson) [inline src/Blocks/Bookmark.php:38]: I solved the unkilled mutation here by throwing a named exception and setting the message in there. The change is repeated for most of the `Blocks` and their tests.

**Comment by mariosimao** [comment]
First of all, thank you very much for your contribution!

I have changed the target branch to `develop` so we can merge everything on `main` when v0.3.0 is ready to be released.

File: infection.json (hunk 1 above)
Comment (mariosimao) [reply infection.json:15]: Maybe in the future, we could increase it to a number close to the actual results. I think it is OK by now.

Comment (WillGibson) [reply infection.json:15]: I think you could move the numbers up, make it red, pick off some mutations and make it green again.

**Comment by WillGibson** [comment]
@mariosimao I meant to add something to the README about the mutation testing. Or maybe it could go in a CONTRIBUTING.md? Do you have a preference?

**Comment by mariosimao** [comment]
Oops sorry for merging too early. I would prefer a separate CONTRIBUTING.md file since not everyone who accesses the repository will contribute.

File: infection.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
+    "$schema": "vendor/infection/infection/resources/schema.json",
+    "source": {
+        "directories": [
+            "src"
+        ]
+    },
+    "logs": {
+        "text": ".infection/infection.log",
```
Comment (Ocramius) [inline infection.json:9]: Suggestion: `"github": true` too - that will give you some decent github output in future :D

**Pull request merged**